File: codility/lessons/2_arrays/cyclic_rotation.cpp
// https://app.codility.com/programmers/lessons/2-arrays/cyclic_rotation/

#include <vector>
using namespace std;

vector<int> rotate(const vector<int> &arr, int k) {
    auto size = arr.size();
    auto rotated = vector<int>(size);

    for (size_t index = 0; index < size; index++) {
        size_t shifted_index = (index + k) % size;
        rotated[shifted_index] = arr[index];
    }

    return rotated;
}

vector<int> solution(vector<int> &A, int K) {
    return rotate(A, K);
}

File: codility/lessons/2_arrays/cyclic_rotation.py
# https://app.codility.com/programmers/lessons/2-arrays/cyclic_rotation/

def rotated(arr, k):
    if len(arr) == 0:
        return arr
    k %= len(arr)
    rotated = arr[-k:] + arr[:-k]
    return rotated

def solution(A, K):
    return rotated(A, K)
